Multiple output formats for generated images

rotarur opened the pull request:
Why do we need this PR?

This PR will introduce multiple formats for the generated images by plantuml as long as update the version of the plantuml.jar file used

What changed?

Added new formats options to entrypoints.sh script
added new option to README.md
added OUTPUT_FORMAT option to actions.yaml file

rotarur committed:
added PNG, SVG, PDF or EPS output formats
updated README.md with the new options
added to actions the option to select the output format for the image

rotarur committed:
added OUTPUT_FORMATS to actions.yaml

rotarur commented:
@frederikheld could you check it please. I would like to merge this to master, later i have some other improvements to do. thanks

frederikheld commented:
Hey @rotarur, thank you very much for this PR :-) I'll look into this over the weekend :-) Cheers, Fred

frederikheld commented on entrypoint.sh:
```diff
@@ -91,9 +117,9 @@ if git commit -m"Auto-generated PlantUML diagrams"; then
-    echo "   ERROR: Could not push to repo."
-    exit 1
-fi
+        echo "   ERROR: Could not push to repo."
```
Thanks for fixing the indentation :-)

rotarur replied:
no problem

frederikheld commented on entrypoint.sh:
```diff
@@ -34,14 +39,35 @@ git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
 
 echo "=> Downloading PlantUML Java app ..."
-wget --quiet -O plantuml.jar https://sourceforge.net/projects/plantuml/files/plantuml.1.2020.15.jar/download
+wget --quiet -O plantuml.jar "$local_plantuml_jar"
```
Great idea to put the link in a variable to avoid hard coded information

rotarur replied:
and to have only one place to change it when needed

frederikheld commented on entrypoint.sh:
```diff
@@ -34,14 +39,35 @@ git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+    echo "=> Preparing output dir ..."
+    mkdir -p "$local_output_dir"
+else
+    echo "=> Output dir $local_output_dir already exists ..."
```
Does this improve user experience? From a technical POV it's not necessary as `mkdir -p` won't throw an error if the directory already exists. And IMHO the information about the output dir isn't crucial here as the user will get the full link to the generated files at the end of the script. My intention with outputs like this is to give an indication what went wrong if the script fails... But I'm also okay with merging the change.

rotarur replied:
you're right. I can remove it in the next code iteration if you want. it doesn't improve the user experience or the performance. I can add debug mode later

frederikheld reviewed (commented):
Cool, looks great :-) Thank you very much! I left you some notes in the code.

From looking at the code I would say this can be merged as it is. However I had no opportunity to test it live yet. I set up a [test repo](https://github.com/frederikheld/test-render-plantuml-to-wiki-action) but I couldn't figure out how to add an action to a workflow from a specific branch in a repo (instead of a specific version in the marketplace). Do you have any idea?

Otherwise I will merge this and then make a 1.1.0-alpha1 release to test with... If this works well, I will release as 1.1.0.

rotarur commented:
can you merge it to the master @frederikheld ? I can't do it. thanks

rotarur commented:
> Cool, looks great :-) Thank you very much! I left you some notes in the code.
> 
> From looking at the code I would say this can be merged as it is. However I had no opportunity to test it live yet. I set up a [test repo](https://github.com/frederikheld/test-render-plantuml-to-wiki-action) but I couldn't figure out how to add an action to a workflow from a specific branch in a repo (instead of a specific version in the marketplace). Do you have any idea?
> 
> Otherwise I will merge this and then make a 1.1.0-alpha1 release to test with... If this works well, I will release as 1.1.0.

isn't the same for branch name? in git a branch name or a tag are the same thing, both can be referenced as a string. create a branch and then try with `frederikheld/render-plantuml-to-wiki-action@MYBRANCH`

frederikheld commented:
You're right, I found it in the docs: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions :-)

I have set up a test workflow in a separate repo (I'm going to try to add this as automated tests into this repo later) where I set up some steps to test image generation:

Workflow script: https://github.com/frederikheld/test-render-plantuml-to-wiki-action/blob/main/.github/workflows/render-plantuml.yml
Successful run: https://github.com/frederikheld/test-render-plantuml-to-wiki-action/actions/runs/1412101542
Wiki embeds: https://github.com/frederikheld/test-render-plantuml-to-wiki-action/wiki

I found an issue with PDF export, which is not working due to missing dependencies. See https://github.com/frederikheld/test-render-plantuml-to-wiki-action/runs/4079460565?check_suite_focus=true#step:7:13.

This seems to be a known issue aka non-feature with PlantUML. See: https://plantuml.com/de/pdf

I suggest to not add PDF export to the action as the action is intended to generate files that can be directly embedded into the wiki (PNG, SVG, JPG, ...). For the same reason (and because it is an ancient format that I see nobody use anymore) I would suggest to not add EPS as well. I'm surprised on the other hand that PlantUML doesn't support JPG. Conversion from PNG to JPG could be done with ImageMagick, but this doesn't need to be covered with this PR.

If PDF and EPS is of use for you, I would suggest to submit it in a separate PR as it needs some deeper changes. E. g. for those file types the action should not generate embed codes but download links (I tried download links [here](https://github.com/frederikheld/test-render-plantuml-to-wiki-action/wiki) if you want to have a look).

Another issue I have found is how the embed markup is generated when the action is used in multiple steps. Apparently my implementation was bad before, but now that different output formats can be created, it shows up as an issue. But it's unrelated to your PR and I have to dig deeper on that.

rotarur committed:
removed pdf and eps formats from output
removed unused if statement
updated README.md file

rotarur commented:
Thanks for the feedback @frederikheld . I've removed the PDF and EPS formats. I don't use them but added as I thought that someone would want to generate a PDF to share with someone else the diagrams.
Also I've removed the if statement that's not used.

Please review

rotarur commented:
@frederikheld could you review this please?

rotarur committed:
added missing required parameter to actions

rotarur commented:
@frederikheld sorry but I'll move out of this project and create my own because I don't see your support on this. it's been 8 days since your last reply. I have an urgency on this matter

frederikheld commented:
Lol, whatever floats your boat.

My last comment <- 6 days -> your last comment <- 2 days -> Conversation is not fast enough for you.

I'm doing this in my free time, if you have time constraints, you can either pay me by the hour to raise priority or invest your own time in your own project 😉

Good luck!

rotarur commented:
my last commit was yesterday and last comment also. i see that you don't have time for this, me either. If you want add me as a contributor to this project and I can use and improve it.

Yes I have some time constraints but I also want to help and contribute. Paying you i wont of course like no one is paying me.

Don't take personal these things, this is how the community is working. if you want to be faster add me as a contributor.
